bumble/profiles/pacs.py

```diff
@@ -0,0 +1,464 @@
+        SAMPLE_FREQUENCY     = 0x01
+        FRAME_DURATION       = 0x02
+        AUDIO_CHANNEL_COUNT  = 0x03
+        OCTETS_PER_SAMPLE    = 0x04
+        CODEC_FRAMES_PER_SDU = 0x05
+
+    supported_sample_frequencies: SupportedSampleFrequency
+    supported_frame_durations: SupportedFrameDuration
+    supported_audio_channel_counts: Sequence[int]
+    min_octets_per_sample: int
+    max_octets_per_sample: int
+    supported_max_codec_frames_per_sdu: int
+
+    @classmethod
+    def from_bytes(cls, data: bytes) -> CodecSpecificCapabilities:
+        pos = 0
+        while pos < len(data):
+            length, type = struct.unpack_from('BB', data, pos)
+            pos += 2
+            value = int.from_bytes(data[pos : pos + length - 1], 'little')
+            pos += length - 1
+
+            if type == CodecSpecificCapabilities.Type.SAMPLE_FREQUENCY:
+                supported_sample_frequencies = SupportedSampleFrequency(value)
+            elif type == CodecSpecificCapabilities.Type.FRAME_DURATION:
+                supported_frame_durations = SupportedFrameDuration(value)
+            elif type == CodecSpecificCapabilities.Type.AUDIO_CHANNEL_COUNT:
+                supported_audio_channel_counts = bits_to_channel_counts(value)
+            elif type == CodecSpecificCapabilities.Type.OCTETS_PER_SAMPLE:
+                min_octets_per_sample = value & 0xFFFF
+                max_octets_per_sample = value >> 16
+            elif type == CodecSpecificCapabilities.Type.CODEC_FRAMES_PER_SDU:
+                supported_max_codec_frames_per_sdu = value
+
+        # It is expected here that if some fields are missing, an error should be raised.
+        return CodecSpecificCapabilities(
+            supported_sample_frequencies=supported_sample_frequencies,
+            supported_frame_durations=supported_frame_durations,
+            supported_audio_channel_counts=supported_audio_channel_counts,
+            min_octets_per_sample=min_octets_per_sample,
+            max_octets_per_sample=max_octets_per_sample,
+            supported_max_codec_frames_per_sdu=supported_max_codec_frames_per_sdu,
+        )
+
+    def __bytes__(self) -> bytes:
+        return struct.pack(
+            '<BBHBBBBBBBBHHBBB',
+            3,
+            CodecSpecificCapabilities.Type.SAMPLE_FREQUENCY,
+            self.supported_sample_frequencies,
+            2,
+            CodecSpecificCapabilities.Type.FRAME_DURATION,
+            self.supported_frame_durations,
+            2,
+            CodecSpecificCapabilities.Type.AUDIO_CHANNEL_COUNT,
+            channel_counts_to_bits(self.supported_audio_channel_counts),
+            5,
+            CodecSpecificCapabilities.Type.OCTETS_PER_SAMPLE,
+            self.min_octets_per_sample,
+            self.max_octets_per_sample,
+            2,
+            CodecSpecificCapabilities.Type.CODEC_FRAMES_PER_SDU,
+            self.supported_max_codec_frames_per_sdu,
+        )
+
+
+@dataclasses.dataclass
+class PAC_Record:
```

Comment:
maybe `PacRecord` to keep a consistent class naming?

Reply:
Done.